theseus/core/theseus_function.py

```diff
@@ -80,6 +88,13 @@ def copy(
                 new_var.name = old_var.name
             for old_aux, new_aux in zip(self.aux_vars, new_fn.aux_vars):
                 new_aux.name = old_aux.name
+
+        if self._has_duplicate_vars(new_fn):
+            raise RuntimeError(
```

Comment:
Is the desired behavior to prevent reuse altogether or should we raise a warning instead and provide some other default behavior.

Could the error message be a bit cryptic, if it get's triggered from some leaf cost function?

Reply:
I think this should be a full error. We were already implicit asking for this contract to be guaranteed by all subclasses (see [this](https://github.com/facebookresearch/theseus/blob/main/theseus/core/theseus_function.py#L67) comment), but we never enforced it in the code. 

I can make the message a bit more helpful, by saying which class should be fixed.

Reply:
Ah okay. Yes the update message sounds good.

Reply:
Added.

Reply:
Nice reads much clearer!